File: benchmarks/phase0/corpus/typescript/flows.ts
function alternateSource(): unknown {
  const expression = process.env.EXPRESSION;
  return eval(expression); // phase0-expect vuln
}

function directSource(request: Request): unknown {
  const expression = request.query.expression;
  return eval(expression); // phase0-expect vuln
}

function constantSink(request: Request): unknown {
  request.query.ignored;
  return eval("2 + 2"); // phase0-expect safe
}

File: benchmarks/phase0/corpus/python/flows.py
import os


def alternate_source() -> object:
    expression = os.getenv("EXPRESSION")
    return eval(expression)  # phase0-expect vuln


def direct_source() -> object:
    expression = input("expression: ")
    return eval(expression)  # phase0-expect vuln


def constant_sink() -> object:
    input("ignored: ")
    return eval("2 + 2")  # phase0-expect safe
